misc/python/materialize/data_ingest/workload.py

```diff
@@ -0,0 +1,152 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import random
+import time
+from copy import deepcopy
+from typing import Any, Dict, List
+
+import pg8000
+
+from materialize.data_ingest.data_type import (
+    DoubleType,
+    FloatType,
+    IntType,
+    LongType,
+    StringType,
+)
+from materialize.data_ingest.definition import (
+    Definition,
+    Delete,
+    Insert,
+    Keyspace,
+    Records,
+    RecordSize,
+    Upsert,
+)
+from materialize.data_ingest.executor import PgExecutor, PrintExecutor
+from materialize.data_ingest.field import Field
+from materialize.data_ingest.transaction import Transaction
+
+
+class Workload:
+    cycle: List[Definition]
+    num_cycles: int
+
+    def generate(self, fields: List[Field]) -> List[Transaction]:
+        transactions = []
+        for i in range(self.num_cycles):
+            cycle = deepcopy(self.cycle)
+            for definition in cycle:
+                transactions.extend(definition.generate(fields))
+        return transactions
+
+
+class SingleSensorUpdating(Workload):
+    def __init__(self, scale_factor: int) -> None:
+        self.num_cycles = 10 * scale_factor
+        self.cycle: List[Definition] = [
+            Upsert(
+                keyspace=Keyspace.SINGLE_VALUE,
+                count=Records.ONE,
+                record_size=RecordSize.SMALL,
+            )
+        ]
+
+
+class DeleteDataAtEndOfDay(Workload):
+    def __init__(self, scale_factor: int) -> None:
+        self.num_cycles = scale_factor
+        insert = Insert(
+            count=Records.ONE,
+            record_size=RecordSize.SMALL,
+        )
+        self.cycle: List[Definition] = [
+            insert,
+            Delete(
+                number_of_records=Records.ALL,
+                record_size=RecordSize.SMALL,
+                num=insert.max_key(),
+            ),
+        ]
+
+
+class ProgressivelyEnrichRecords(Workload):
+    def __init__(self, scale_factor: int) -> None:
+        self.num_cycles = scale_factor
+        self.cycle: List[Definition] = [
+            # TODO: Implement
+        ]
+
+
+# TODO: Disruptions in workloads
+
+
+def execute_workload(
+    executor_classes: Any,
+    workload: Workload,
+    conn: pg8000.Connection,
+    num: int,
+    ports: Dict[str, int],
+    verbose: bool,
+) -> None:
+    fields = []
+
+    for i in range(random.randint(1, 10)):
+        typ = random.choice((StringType, IntType, LongType, FloatType, DoubleType))
```

Comment:
Are these all types? Isn't at least `BooleanType` missing?
Introduce a constant instead of redefining the tuple a few lines later again? (This will reduce the risk that one forgets to update one of the occurrences when introducing a new type.)

Reply:
Those are just the ones we want to use for now. As Philip noted, probably don't need all types.